flang/include/flang/Parser/parse-tree.h

```diff
@@ -3440,6 +3440,16 @@ struct OmpObject {
 
 WRAPPER_CLASS(OmpObjectList, std::list<OmpObject>);
 
+#define MODIFIER_BOILERPLATE(...) \
+  struct Modifier { \
+    using Variant = std::variant<__VA_ARGS__>; \
+    UNION_CLASS_BOILERPLATE(Modifier); \
+    CharBlock source; \
+    Variant u; \
+  }
+
+#define MODIFIERS() std::optional<std::list<Modifier>>
```

Comment:
nit: why is this a macro rather than a type alias? It is okay if you have a good reason, it was just surprising for me to read.

Reply:
Because `Modifier` is not defined at this point.  It's a local class that each clause defines individually.

Reply:
You could do something like
```c++
template <typename MODIFIER>
using Modifiers = std::optional<std::list<MODIFIER>>;
```

Reply:
The purpose of the `MODIFIERS()` macro is to hide the name "Modifier" that the boilerplate macro introduces.

The `MODIFIER_BOILERPLATE(type1, type2, ...)` macro expands to (roughly)
```
struct Modifier {
  std::variant<type1, type2, ...> u;
};
```
So inside a clause class it will be
```
struct OmpSomeClause {
  struct Modifier {
    std::variant<...> u;
  };
  std::tuple<std::optional<std::list<Modifier>>, ClauseArg1, ClauseArg2...> t;
};
```
The `std::optional<std::list<Modifier>>` here is what the `MODIFIERS()` macro inserts.  It's always the same string (hence no arguments to the macro), but the "Modifier" word in it refers to a different thing each time.

With the alias you suggested, the user would always have to say
```
struct OmpSomeClause {
   MODIFIER_BOILERPLATE(type1, type2, ...)
   std::tuple<Modifiers<Modifier>, ClauseArg1, ClauseArg2, ...> t;
};
```
and explicitly use the word "Modifier".

Reply:
Ahh I understand now. Thanks for explaining. Yes this makes sense to me as it is.